browser/ui/views/sidebar/sidebar_show_options_event_detect_widget.cc

```diff
@@ -78,26 +78,21 @@ class SidebarShowOptionsEventDetectWidget::ContentsView : public views::View {
   }
 
   std::u16string GetTooltipText(const gfx::Point& p) const override {
-    if (show_on_hover_)
+    if (delegate_->ShouldShowOnHover())
       return View::GetTooltipText(p);
 
     return l10n_util::GetStringUTF16(
         IDS_SIDEBAR_TOOLTIP_ON_SHOW_OPTIONS_WIDGET);
   }
 
-  void set_show_on_hover(bool show_on_hover) { show_on_hover_ = show_on_hover; }
-
  private:
-  // true for showing sidebar on hover. false for showing sidebar on hover and
-  // click this area.
-  bool show_on_hover_ = false;
   SidebarShowOptionsEventDetectWidget::Delegate* delegate_ = nullptr;
 };
 
 SidebarShowOptionsEventDetectWidget::SidebarShowOptionsEventDetectWidget(
     BraveBrowserView* browser_view,
     Delegate* delegate)
-    : browser_view_(browser_view) {
+    : browser_view_(browser_view), delegate_(delegate) {
   observation_.Observe(browser_view_->contents_container());
```

Comment:
Should we add a `DCHECK(delegate_)` here? I noticed we reference it without a nullptr check in a few places.

Reply:
Yup, it should be checked. Added. Thanks!